clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAO.scala

```diff
@@ -62,31 +62,16 @@ class HttpElasticsearchDAO private[dataaccess] (
     searchResponse flatMap foldScroll(Seq.empty[D], searchDefinition.keepAlive)
   }
 
-  /**
-    * Given an elastic search index, return the most recent document for that index.
-    *
-    * Document ordering is determined by the "upsert ID", which must be indexed field.
-    *
-    * @param index the elasticsearch index
-    * @tparam D the document type this index contains
-    * @return the most recent document for this index, if any
-    */
-  def getMostRecentDocument[D <: ClioDocument: HitReader](
+  override def getMostRecentDocument[D <: ClioDocument: HitReader](
     index: ElasticsearchIndex[D]
-  ): Future[D] = {
+  ): Future[Option[D]] = {
     val searchDefinition = search(index.indexName / index.indexType)
       .size(1)
       .sortByFieldDesc(ClioDocument.UpsertIdElasticSearchName)
     for {
       searchResponse <- httpClient execute searchDefinition
     } yield {
-      searchResponse.to[D] match {
-        case Seq(document) => document
-        case _ =>
-          throw new RuntimeException(
```

Comment:
Does this still result in an exception being thrown if the `execute` command fails? E.g. can the caller distinguish an error case from the case where no documents are present?

Reply:
If `execute` fails, this whole `yield` block won't run and the caller will get back a failed `Future` containing the exception.